Bot fails to start after update to v1.3.0

<!-- Complete the necessary portions of this template and delete the rest. -->

## Describe the bug

After updating to v1.3.0 the bot does not start in the emulator. 
![image](https://user-images.githubusercontent.com/76207835/102606407-fd599000-4126-11eb-8d75-0a49e292b15e.png)


Publishing to azure is still going fine
![image](https://user-images.githubusercontent.com/76207835/102606351-e74bcf80-4126-11eb-9f2b-38964e576d56.png)



## Version

1.3.0

## Browser

<!-- What browser are you using? -->

- [ ] Electron distribution
- Chrome
- [ ] Safari
- [ ] Firefox
- Edge

## OS

<!-- What operating system are you using? -->

- [ ] macOS
- Windows
- [ ] Ubuntu

## To Reproduce

Steps to reproduce the behavior:

1. Open bot
2. Click on start bot or restart bot

## Expected behavior

Bot opens in emulator

## Screenshots

<!-- If applicable, add screenshots/gif/video to help explain your problem. -->

## Additional context

<!-- Add any other context about the problem here. -->

@luhan2017 I followed your steps and tried connecting manually with the emulator and got this exeption:

PS C:\Users\snoekya\Desktop\Stage\BotComposer_project\Isabot\runtime> dotnet run --project azurewebapp
C:\Users\snoekya\Desktop\Stage\BotComposer_project\Isabot\runtime\azurewebapp\Microsoft.BotFramework.Composer.WebApp.csproj : warning
 NU1701: Package 'Microsoft.Azure.KeyVault.Core 1.0.0' was restored using '.NETFramework,Version=v4.6.1, .NETFramework,Version=v4.6.2
, .NETFramework,Version=v4.7, .NETFramework,Version=v4.7.1, .NETFramework,Version=v4.7.2, .NETFramework,Version=v4.8' instead of the 
project target framework '.NETCoreApp,Version=v3.1'. This package may not be fully compatible with your project.
C:\Users\snoekya\Desktop\Stage\BotComposer_project\Isabot\runtime\core\Microsoft.BotFramework.Composer.Core.csproj : warning NU1701: 
Package 'Microsoft.Azure.KeyVault.Core 1.0.0' was restored using '.NETFramework,Version=v4.6.1, .NETFramework,Version=v4.6.2, .NETFra
mework,Version=v4.7, .NETFramework,Version=v4.7.1, .NETFramework,Version=v4.7.2, .NETFramework,Version=v4.8' instead of the project t
arget framework '.NETStandard,Version=v2.0'. This package may not be fully compatible with your project. [C:\Users\snoekya\Desktop\St
age\BotComposer_project\Isabot\runtime\azurewebapp\Microsoft.BotFramework.Composer.WebApp.csproj]
C:\Users\snoekya\Desktop\Stage\BotComposer_project\Isabot\runtime\azurewebapp\Microsoft.BotFramework.Composer.WebApp.csproj : warning
 NU1701: Package 'Microsoft.Azure.KeyVault.Core 1.0.0' was restored using '.NETFramework,Version=v4.6.1, .NETFramework,Version=v4.6.2
, .NETFramework,Version=v4.7, .NETFramework,Version=v4.7.1, .NETFramework,Version=v4.7.2, .NETFramework,Version=v4.8' instead of the 
project target framework '.NETCoreApp,Version=v3.1'. This package may not be fully compatible with your project.
C:\Users\snoekya\Desktop\Stage\BotComposer_project\Isabot\runtime\core\Microsoft.BotFramework.Composer.Core.csproj : warning NU1701: 
Package 'Microsoft.Azure.KeyVault.Core 1.0.0' was restored using '.NETFramework,Version=v4.6.1, .NETFramework,Version=v4.6.2, .NETFra
mework,Version=v4.7, .NETFramework,Version=v4.7.1, .NETFramework,Version=v4.7.2, .NETFramework,Version=v4.8' instead of the project t
arget framework '.NETStandard,Version=v2.0'. This package may not be fully compatible with your project.
info: Microsoft.Hosting.Lifetime[0]
      Now listening on: http://localhost:3979
info: Microsoft.Hosting.Lifetime[0]
      Application started. Press Ctrl+C to shut down.
info: Microsoft.Hosting.Lifetime[0]
      Hosting environment: Development
info: Microsoft.Hosting.Lifetime[0]
      Content root path: C:\Users\snoekya\Desktop\Stage\BotComposer_project\Isabot\runtime\azurewebapp
info: Microsoft.AspNetCore.Hosting.Diagnostics[1]
      Request starting HTTP/1.1 POST http://localhost:3979/api/messages application/json 626
info: Microsoft.AspNetCore.Routing.EndpointMiddleware[0]
      Executing endpoint 'Microsoft.BotFramework.Composer.WebAppTemplates.Controllers.BotController.PostAsync (Microsoft.BotFramework
.Composer.WebApp)'
info: Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker[3]
      Route matched with {action = "Post", controller = "Bot"}. Executing controller action with signature System.Threading.Tasks.Tas
k PostAsync() on controller Microsoft.BotFramework.Composer.WebAppTemplates.Controllers.BotController (Microsoft.BotFramework.Compose
r.WebApp).
info: Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker[1]
      Executing action method Microsoft.BotFramework.Composer.WebAppTemplates.Controllers.BotController.PostAsync (Microsoft.BotFrame
work.Composer.WebApp) - Validation state: Valid
info: Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker[2]
      Executed action Microsoft.BotFramework.Composer.WebAppTemplates.Controllers.BotController.PostAsync (Microsoft.BotFramework.Com
poser.WebApp) in 922.6087ms
info: Microsoft.AspNetCore.Routing.EndpointMiddleware[1]
      Executed endpoint 'Microsoft.BotFramework.Composer.WebAppTemplates.Controllers.BotController.PostAsync (Microsoft.BotFramework.
Composer.WebApp)'
fail: Microsoft.AspNetCore.Server.Kestrel[13]
      Connection id "0HM5BT3PP82NS", Request id "0HM5BT3PP82NS:00000001": An unhandled exception was thrown by the application.
System.ArgumentNullException: Value cannot be null. (Parameter 'clientSecret')
   at Microsoft.IdentityModel.Clients.ActiveDirectory.ClientCredential..ctor(String clientId, String clientSecret)
   at Microsoft.Bot.Connector.Authentication.MicrosoftAppCredentials.<BuildAuthenticator>b__14_0()
   at System.Lazy`1.ViaFactory(LazyThreadSafetyMode mode)
   at System.Lazy`1.ExecutionAndPublication(LazyHelper executionAndPublication, Boolean useDefaultConstructor)
   at System.Lazy`1.CreateValue()
   at System.Lazy`1.get_Value()
   at Microsoft.Bot.Connector.Authentication.AppCredentials.<BuildIAuthenticator>b__36_0()
   at System.Lazy`1.ViaFactory(LazyThreadSafetyMode mode)
--- End of stack trace from previous location where exception was thrown ---
   at System.Lazy`1.CreateValue()
   at System.Lazy`1.get_Value()
   at Microsoft.Bot.Connector.Authentication.AppCredentials.GetTokenAsync(Boolean forceRefresh)
   at Microsoft.Bot.Connector.Authentication.AppCredentials.ProcessHttpRequestAsync(HttpRequestMessage request, CancellationToken can
cellationToken)
   at Microsoft.Bot.Connector.Conversations.ReplyToActivityWithHttpMessagesAsync(String conversationId, String activityId, Activity a
ctivity, Dictionary`2 customHeaders, CancellationToken cancellationToken)
   at Microsoft.Bot.Connector.ConversationsExtensions.ReplyToActivityAsync(IConversations operations, String conversationId, String a
ctivityId, Activity activity, CancellationToken cancellationToken)
   at Microsoft.Bot.Builder.BotFrameworkAdapter.SendActivitiesAsync(ITurnContext turnContext, Activity[] activities, CancellationToke
n cancellationToken)
   at Microsoft.Bot.Builder.TurnContext.<>c__DisplayClass30_0.<<SendActivitiesAsync>g__SendActivitiesThroughAdapter|1>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.Bot.Builder.TurnContext.SendActivityAsync(IActivity activity, CancellationToken cancellationToken)
   at Microsoft.Bot.Builder.TurnContext.SendActivityAsync(String textReplyToSend, String speak, String inputHint, CancellationToken c
ancellationToken)
   at Microsoft.BotFramework.Composer.WebAppTemplates.Startup.<>c__DisplayClass12_0.<<GetBotAdapter>b__0>d.MoveNext() in C:\Users\sno
ekya\Desktop\Stage\BotComposer_project\Isabot\runtime\azurewebapp\Startup.cs:line 122
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.Bot.Builder.BotAdapter.RunPipelineAsync(ITurnContext turnContext, BotCallbackHandler callback, CancellationToken canc
ellationToken)
   at Microsoft.Bot.Builder.BotFrameworkAdapter.ProcessActivityAsync(ClaimsIdentity claimsIdentity, Activity activity, BotCallbackHan
dler callback, CancellationToken cancellationToken)
   at Microsoft.Bot.Builder.BotFrameworkAdapter.ProcessActivityAsync(String authHeader, Activity activity, BotCallbackHandler callbac
k, CancellationToken cancellationToken)
   at Microsoft.Bot.Builder.Integration.AspNet.Core.BotFrameworkHttpAdapter.ProcessAsync(HttpRequest httpRequest, HttpResponse httpRe
sponse, IBot bot, CancellationToken cancellationToken)
   at Microsoft.BotFramework.Composer.WebAppTemplates.Controllers.BotController.PostAsync() in C:\Users\snoekya\Desktop\Stage\BotComp
oser_project\Isabot\runtime\azurewebapp\Controllers\BotController.cs:line 38
   at lambda_method(Closure , Object )
   at Microsoft.Extensions.Internal.ObjectMethodExecutorAwaitable.Awaiter.GetResult()
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.AwaitableResultExecutor.Execute(IActionResultTypeMapper mapper, Ob
jectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Logged|12_1(ControllerActionInvoker
 invoker)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionIn
voker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompl
eted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeInnerFilterAsync>g__Awaited|13_0(ControllerActionInvoker
 invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|19_0(ResourceInvoker invoker, Tas
k lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Logged|17_1(ResourceInvoker invoker)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger lo
gger)
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpProtocol.ProcessRequests[TContext](IHttpApplication`1 application)
info: Microsoft.AspNetCore.Hosting.Diagnostics[2]
      Request finished in 1010.8007ms 500

--- a/extensions/localPublish/src/index.ts
+++ b/extensions/localPublish/src/index.ts
@@ -53,12 +53,10 @@ const isSkillHostUpdateRequired = (skillHostEndpoint?: string) => {
 };
 
 const stringifyError = (error: any): string => {
-  if (error instanceof Error) {
-    return error.toString();
-  } else if (typeof error === 'object') {
+  if (typeof error === 'object') {
     return JSON.stringify(error, Object.getOwnPropertyNames(error));
   } else {
-    return error;
+    return error.toString();
   }
 }
 

--- a/extensions/localPublish/src/index.ts
+++ b/extensions/localPublish/src/index.ts
@@ -54,7 +54,7 @@ const isSkillHostUpdateRequired = (skillHostEndpoint?: string) => {
 
 const stringifyError = (error: any): string => {
   if (error instanceof Error) {
-    return `${error.name} - ${error.message}`;
+    return error.toString();
   } else if (typeof error === 'object') {
     return JSON.stringify(error, Object.getOwnPropertyNames(error));
   } else {

--- a/extensions/localPublish/src/index.ts
+++ b/extensions/localPublish/src/index.ts
@@ -16,7 +16,6 @@ import { DialogSetting, PublishPlugin, IExtensionRegistration } from '@botframew
 
 import killPort from 'kill-port';
 import map from 'lodash/map';
-import range from 'lodash/range';
 import * as tcpPortUsed from 'tcp-port-used';
 
 const stat = promisify(fs.stat);
@@ -52,6 +51,17 @@ const isLocalhostUrl = (matchUrl: string) => {
 const isSkillHostUpdateRequired = (skillHostEndpoint?: string) => {
   return !skillHostEndpoint || isLocalhostUrl(skillHostEndpoint);
 };
+
+const stringifyError = (error: any): string => {
+  if (error instanceof Error) {
+    return `${error.name} - ${error.message}`;
+  } else if (typeof error === 'object') {
+    return JSON.stringify(error, Object.getOwnPropertyNames(error));
+  } else {
+    return error;
+  }
+}
+
 class LocalPublisher implements PublishPlugin<PublishConfig> {
   public name = 'localpublish';
   public description = 'Publish bot to local runtime';
@@ -114,7 +124,7 @@ class LocalPublisher implements PublishPlugin<PublishConfig> {
       this.setBotStatus(botId, {
         status: 500,
         result: {
-          message: error.message,
+          message: stringifyError(error),
         },
       });
     }
@@ -149,7 +159,7 @@ class LocalPublisher implements PublishPlugin<PublishConfig> {
       return {
         status: 500,
         result: {
-          message: error,
+          message: stringifyError(error),
         },
       };
     }
@@ -279,7 +289,7 @@ class LocalPublisher implements PublishPlugin<PublishConfig> {
     } catch (error) {
       // delete the folder to make sure build again.
       await removeDirAndFiles(this.getBotDir(botId));
-      throw new Error(error.toString());
+      throw error;
     }
   };
 
@@ -324,12 +334,11 @@ class LocalPublisher implements PublishPlugin<PublishConfig> {
       // start the bot process
       await this.startBot(botId, port, settings, project);
     } catch (error) {
-      console.error('Error in startbot: ', error);
       await this.stopBot(botId);
       this.setBotStatus(botId, {
         status: 500,
         result: {
-          message: error,
+          message: stringifyError(error),
         },
       });
     }

